src/NCPkgFilterService.cc

```diff
@@ -0,0 +1,310 @@
+| modify it under the terms of version 2 of the GNU General Public License as
+| published by the Free Software Foundation.
+|
+| This program is distributed in the hope that it will be useful,
+| but WITHOUT ANY WARRANTY; without even the implied warranty of
+| MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.   See the
+| GNU General Public License for more details.
+|
+| You should have received a copy of the GNU General Public License
+| along with this program; if not, contact SUSE.
+|
+| To contact SUSE about this file by physical or electronic mail,
+| you may find current contact information at www.suse.com
+|
+|***************************************************************************/
+
+#define YUILogComponent "ncurses-pkg"
+#include <YUILog.h>
+
+#include <zypp/ServiceInfo.h>
+#include <boost/algorithm/string.hpp>
+
+#include "NCPkgFilterService.h"
+
+#include "YDialog.h"
+#include "NCLayoutBox.h"
+#include "NCSpacing.h"
+#include "NCPackageSelector.h"
+
+#include "NCZypp.h"
+
+using std::endl;
+
+/*
+  Textdomain "ncurses-pkg"
+*/
+
+///////////////////////////////////////////////////////////////////
+//
+//
+//	METHOD NAME : NCPkgServiceTag::NCPkgServiceTag
+//	METHOD TYPE : Constructor
+//
+//	DESCRIPTION :
+//
+
+NCPkgServiceTag::NCPkgServiceTag ( ZyppService servicePtr)
+    : YTableCell(std::string(" "))
+    , service (servicePtr)
+{
+
+}
+
+///////////////////////////////////////////////////////////////////
+//
+//
+//	METHOD NAME : NCPkgServiceTable::NCPkgServiceTable
+//	METHOD TYPE : Constructor
+//
+//	DESCRIPTION :
+//
+
+NCPkgServiceTable::NCPkgServiceTable( YWidget *parent, YTableHeader *tableHeader, NCPackageSelector *pkg )
+    :NCTable( parent, tableHeader )
+    ,packager(pkg)
+    ,repo_manager(new zypp::RepoManager())
+{
+   fillHeader();
+   fillServiceList();
+}
+
+bool NCPkgServiceTable::any_service()
+{
+    bool ret = std::any_of(ZyppRepositoriesBegin(), ZyppRepositoriesEnd(), [](const zypp::Repository& repo) {
+        // if the repository does not belong to any service then the service name is empty
+        return !repo.info().service().empty();
+    });
+
+    yuiMilestone() << "Found a libzypp service: " << ret << std::endl;
+    return ret;
+}
+
+///////////////////////////////////////////////////////////////////
+//
+//
+//	METHOD NAME : NCPkgServiceTable::fillHeader
+//	METHOD TYPE : void
+//
+//	DESCRIPTION : Fill header of servicesitories table (name + URL)
+//
+
+void NCPkgServiceTable::fillHeader()
+{
+    std::vector <std::string> header;
+
+    header.reserve(2);
+    header.push_back( "L" );
+    header.push_back( "L" + NCPkgStrings::PkgName() );
+
+    setHeader( header);
+}
+
+///////////////////////////////////////////////////////////////////
+//
+//
+//	METHOD NAME : NCPkgServiceTable::addLine
+//	METHOD TYPE : void
+//
+//	DESCRIPTION : Add one line (with tag) to the service table
+//
+
+void NCPkgServiceTable::addLine ( ZyppService r, const std::vector <std::string> & cols )
```

Comment:
NP: The parameter name `r` is probably a leftover from the repo filter view where it implied `repo`, so it's kind of confusing here.